### PR TITLE
[0366] Allow course code to be updated in the support UI

### PR DESCRIPTION
**Pull request by defong**
### Context
Course code

### Changes proposed in this pull request
Allow course code to be editable

### Guidance to review
It works, only basic validations, its not blank and its unique.
course-code-editing

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [ ] Tested by running locally

### CODE REVIEW

**Comment by defong** [comment]
[Allow course code to be updated in the support UI](https://trello.com/c/4Sdp1QUZ/366-allow-course-code-to-be-updated-in-the-support-ui)

**Review by mbocc — APPROVED** [review]
Great gif. Reviewed, all good and works. Can now easily edit course codes via UI. 
Thank you!!

File: spec/features/support/providers/courses/editing_a_course_spec.rb [hunk 1]
```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+feature "Edit provider course cide" do
```
Comment (malcolmbaig) [inline spec/features/support/providers/courses/editing_a_course_spec.rb:5]:
```suggestion
feature "Edit provider course code" do
```

File: spec/features/support/providers/courses/editing_a_course_spec.rb [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  let(:user) { create(:user, :admin) }
+
+  before do
+    given_i_am_authenticated(user: user)
```
Comment (malcolmbaig) [inline spec/features/support/providers/courses/editing_a_course_spec.rb:9]: Can we replace this with a single step that creates and authenticates the user? Would like to clear up some of the `let` and `before` blocks in these feature specs over time.

```suggestion
    given_i_am_authenticated_as_an_admin_user
```

**Review by malcolmbaig — COMMENTED** [review]
Looks great, couple of minor comments left inline.

**Review by malcolmbaig — APPROVED** [review]

**Pull request merged**